v2/streaming-data-generator/src/test/java/com/google/cloud/teleport/v2/templates/StreamingDataGeneratorIT.java

```diff
@@ -94,4 +128,171 @@ public void testFakeMessagesToGcs() throws IOException {
+            .addParameter(
+                "outputTableSpec",
+                String.format(
+                    "%s:%s.%s", PROJECT, bigQueryResourceManager.getDatasetId(), jobName));
+
+    // Act
+    JobInfo info = launchTemplate(options);
+    assertThat(info.state()).isIn(JobState.ACTIVE_STATES);
+    Result result =
+        new DataflowOperator(getDataflowClient())
+            .waitForConditionAndFinish(
+                createConfig(info),
+                () -> bigQueryResourceManager.readTable(jobName).getTotalRows() > 0);
+    // Assert
+    assertThat(result).isEqualTo(Result.CONDITION_MET);
+  }
+
+  @Test
+  public void testFakeMessagesToSpanner() throws IOException {
+    // Arrange
+    String name = testName.getMethodName();
+    String jobName = createJobName(name);
+    spannerResourceManager =
+        DefaultSpannerResourceManager.builder(testName.getMethodName(), PROJECT, REGION).build();
+    String createTableStatement =
+        String.format(
+            "CREATE TABLE `%s` (\n"
+                + "  eventId STRING(1024) NOT NULL,\n"
+                + "  eventTimestamp INT64,\n"
+                + "  ipv4 STRING(1024),\n"
+                + "  ipv6 STRING(1024),\n"
+                + "  country STRING(1024),\n"
+                + "  username STRING(1024),\n"
+                + "  quest STRING(1024),\n"
+                + "  score INT64,\n"
+                + "  completed BOOL,\n"
+                + ") PRIMARY KEY(eventId)",
+            testName.getMethodName());
+    ImmutableList<String> columnNames =
+        ImmutableList.of(
+            "eventId",
+            "eventTimestamp",
+            "ipv4",
+            "ipv6",
+            "country",
+            "username",
+            "quest",
+            "score",
+            "completed");
+    spannerResourceManager.createTable(createTableStatement);
+
+    LaunchConfig.Builder options =
+        LaunchConfig.builder(jobName, specPath)
+            .addParameter(SCHEMA_TEMPLATE_KEY, SchemaTemplate.GAME_EVENT.name())
+            .addParameter(QPS_KEY, DEFAULT_QPS)
+            .addParameter(SINK_TYPE_KEY, SinkType.SPANNER.name())
+            .addParameter("projectId", PROJECT)
+            .addParameter("spannerInstanceName", spannerResourceManager.getInstanceId())
+            .addParameter("spannerDatabaseName", spannerResourceManager.getDatabaseId())
+            .addParameter("spannerTableName", testName.getMethodName());
+
+    // Act
+    JobInfo info = launchTemplate(options);
+    assertThat(info.state()).isIn(JobState.ACTIVE_STATES);
+
+    Result result =
+        new DataflowOperator(getDataflowClient())
+            .waitForConditionAndFinish(
+                createConfig(info),
+                () ->
+                    !spannerResourceManager
+                        .readTableRecords(testName.getMethodName(), columnNames)
+                        .isEmpty());
+
+    // Assert
+    assertThat(result).isEqualTo(Result.CONDITION_MET);
```

Comment:
This is hard to do since the generated record is fake and created using a very vague schema template. I think the fact that the data exists in the BQ/Spanner table proves the validity of the data (since it has to match the schema of the table)